### ISSUE TITLE
Taskloop tasks running sequentially

### ISSUE THREAD

**Issue by kiranchandramohan** [flang-compiler/flang#406]
omp parallel -> omp single -> omp taskloop is a common idiom used in openmp applications. This causes a single thread to generate all the tasks and then all the threads can work together on the generated tasks. This idiom is used in the code given below. 

Unfortunately, this code when compiled by flang does not run the tasks in parallel. On an investigation it was found that the call to kmpc_taskloop in the flang generated llvm ir has the if_clause parameter value set to 0. This causes all the tasks to be sequential since this setting forces the generated task to be run immediately and not deferred.

program tskloop
  integer, parameter :: n = 10000000
  real :: arr(n)

  !$omp parallel
  !$omp single
  !$omp taskloop default (none) private (i) shared (arr)
  do i = 1, n
     arr(i) = log(sqrt(log((real(i)))**4.5)**10.3)
  end do
  !$omp end taskloop
  !$omp end single
  !$omp end parallel
  print *, arr(n-1)
end

**Comment by kiranchandramohan** [comment]
I see that the following commit fixes this issue. If you (@sscalpone / @gklimowicz ) disagree please reopen.
https://github.com/flang-compiler/flang/commit/ba5322e8814d44bc756df3f26bcad2d4d8eeacb5

**Comment by jeffhammond** [comment]
@kiranchandramohan Commit logs are not a good way to determine if bugs are fixed.  You need to test.  In this case, I ran the test and it fails.

# Failure
```
jrhammon@klondike:~/prk-repo/FORTRAN$ /opt/llvm/pgi-flang/bin/flang -Mpreprocess -Mfreeform -L/opt/llvm/pgi-flang/lib -Wl,-rpath=/opt/llvm/pgi-flang/lib -g -O3 -fopenmp -L/opt/llvm/5.0.1/lib tl.f90 -o tl.x
F90-S-0034-Syntax error at or near identifier loop (tl.f90: 7)
F90-W-0287-Unrecognized OpenMP directive - end  (tl.f90: 11)
  0 inform,   1 warnings,   1 severes, 0 fatal for tskloop
```

# Source
```
$ cat tl.f90
program tskloop
integer, parameter :: n = 10000000
real :: arr(n)

!$omp parallel
!$omp single
!$omp taskloop default (none) private (i) shared (arr)
do i = 1, n
arr(i) = log(sqrt(log((real(i)))**4.5)**10.3)
end do
!$omp end taskloop
!$omp end single
!$omp end parallel
print *, arr(n-1)
end
```

# Compiler
```
$ /opt/llvm/pgi-flang/bin/flang -v
clang version 6.0.1 (https://github.com/flang-compiler/flang-driver.git dd7587310ae498c22514a33e1a2546b86af9cf25) (https://github.com/llvm-mirror/llvm.git 5136df4d089a086b70d452160ad5451861269498)
Target: x86_64-unknown-linux-gnu
Thread model: posix
InstalledDir: /opt/llvm/pgi-flang/bin
Found candidate GCC installation: /usr/lib/gcc/i686-linux-gnu/6
Found candidate GCC installation: /usr/lib/gcc/i686-linux-gnu/6.4.0
Found candidate GCC installation: /usr/lib/gcc/i686-linux-gnu/7
Found candidate GCC installation: /usr/lib/gcc/i686-linux-gnu/7.3.0
Found candidate GCC installation: /usr/lib/gcc/i686-linux-gnu/8
Found candidate GCC installation: /usr/lib/gcc/x86_64-linux-gnu/4.9
Found candidate GCC installation: /usr/lib/gcc/x86_64-linux-gnu/4.9.4
Found candidate GCC installation: /usr/lib/gcc/x86_64-linux-gnu/5
Found candidate GCC installation: /usr/lib/gcc/x86_64-linux-gnu/5.5.0
Found candidate GCC installation: /usr/lib/gcc/x86_64-linux-gnu/6
Found candidate GCC installation: /usr/lib/gcc/x86_64-linux-gnu/6.4.0
Found candidate GCC installation: /usr/lib/gcc/x86_64-linux-gnu/7
Found candidate GCC installation: /usr/lib/gcc/x86_64-linux-gnu/7.3.0
Found candidate GCC installation: /usr/lib/gcc/x86_64-linux-gnu/8
Selected GCC installation: /usr/lib/gcc/x86_64-linux-gnu/7.3.0
Candidate multilib: .;@m64
Candidate multilib: 32;@m32
Candidate multilib: x32;@mx32
Selected multilib: .;@m64
Found CUDA installation: /usr/local/cuda, version unknown
```
